42/README.md

```diff
@@ -70,14 +71,129 @@ The above process ensure the node keeps validating and accepting new blocks whil
+```
+
+Validators can watch how the election goes:
+
+```
+$ bigchaindb migration status <migration-id>
+```
+
+Outputs:
+```
+votes_recieved=<Sum_of_votes_recieved>
+votes_allocated=<Sum_of_votes_allocated_in_election>
+network_size=<Total_network_power>
+```
+
+#### 2. Replay the blockchain
+
+In order to replay the chain up to the migration height, we need to either keep the old Tendermint version running or skip replaying the corresponding part of the chain = archive the chain. We consider the former to be a huge hassle so we conceive an archiving scheme further in this chapter.
+
+#### 3. Start the new chain
+
+Validators have to install and launch the new version of Tendermint. They need to prepare a new `genesis.json`. The new genesis file has to contain the validator set, the application hash (both at the migration height), and the identifier of the new Tendermint chain (`chain_id`).
+
+The application hash is calculated by computing the Merkle tree root of the current UTXO.
+
+`chain_id` is generated and stored by BigchainDB upon conclusion of the migration election. When Tendermint sends this ID as part of the `InitChain` ABCI request, BigchainDB understands that the user has switched to a new Tendermint version so BigchainDB switches to accept new transactions and blocks.
+
+To offer a convenient way to get the data, we propose a CLI command:
+
+```
+$ bigchaindb status
+```
+
+Its output can be extended over time to solve different purposes. For a start, it has to contain the latest known list of validators, the latest known app hash, and the identifier of the Tendermint chain.
+```
+{
+    "chain_id": "...",
+    "validators": [{
+        "pub_key": {
+            "type": "...",
+            "value": "..."
+        },
+        "power": "...",
+        "name": "..."
+    },
+    "app_hash": "..."
+}
+```
+
+Validators upgrading Tendermint are supposed to execute the command above and copy the validators, the app hash, and the chain ID into their `genesis.json`.
+
+Validators joining the network after the migration take the genesis file from existing validators, as usual.
+
+#### 4. Join the network after a migration
+
+When a validator joins the network after a migration, it does not suffice for him to know `genesis.json` - he also needs the archive of the old chain (heights from 0 to the latest migration height).
+
+Such a validator needs to take a dump of the old chain from an existing member of the network. The dump can be created via a CLI command:
+
+```
+$ bigchaindb migration dump
+```
+
+The command creates the `bigchaindb.dump` file that needs to be transferred to the new member. The dump contains all BigchainDB collections to restore the old chain from.
+
+The new validator needs to start BigchainDB and apply the dump:
+
+```
+$ bigchaindb migration apply <dump>
+```
+
+After the dump is applied, Tendermint can be started.
```

Comment:
what happens if tendermint is started before the migration dump is applied? it will still sync with the current state of the new chain. I supposes tendermint can be started before or after the migration dump is applied but the bigchaindb server can only be started after the dump has been applied